src/main/kotlin/com/openlattice/chronicle/controllers/SurveyController.kt

```diff
@@ -0,0 +1,185 @@
+        return OK("Successfully deleted questionnaire $questionnaireId")
+    }
+
+    @GetMapping(
+        path = [STUDY_ID_PATH + QUESTIONNAIRE_PATH + QUESTIONNAIRE_ID_PATH],
+        produces = [MediaType.APPLICATION_JSON_VALUE]
+    )
+    override fun getQuestionnaire(
+        @PathVariable(STUDY_ID) studyId: UUID,
+        @PathVariable(QUESTIONNAIRE_ID) questionnaireId: UUID
+    ): Questionnaire {
+        return surveysService.getQuestionnaire(studyId, questionnaireId)
+    }
+
+    @PatchMapping(
+        path = [STUDY_ID_PATH + QUESTIONNAIRE_PATH + QUESTIONNAIRE_ID_PATH],
+        consumes = [MediaType.APPLICATION_JSON_VALUE],
+    )
+    override fun updateQuestionnaire(
+        @PathVariable(STUDY_ID) studyId: UUID,
+        @PathVariable(QUESTIONNAIRE_ID) questionnaireId: UUID,
+        @RequestBody update: QuestionnaireUpdate
+    ): OK {
+        ensureWriteAccess(AclKey(studyId))
+        surveysService.updateQuestionnaire(studyId, questionnaireId, update)
+
+        return OK("Successfully updated questionnaire $questionnaireId")
+    }
+
+    @GetMapping(
+        path = [STUDY_ID_PATH + QUESTIONNAIRE_PATH],
+        produces = [MediaType.APPLICATION_JSON_VALUE]
+    )
+    override fun getStudyQuestionnaires(@PathVariable(STUDY_ID) studyId: UUID): List<Questionnaire> {
+        return surveysService.getStudyQuestionnaires(studyId)
+    }
+
+    @PostMapping(
+        path = [STUDY_ID_PATH + PARTICIPANT_PATH + PARTICIPANT_ID_PATH + QUESTIONNAIRE_PATH + QUESTIONNAIRE_ID_PATH],
+        produces = []
+    )
+    override fun submitQuestionnaireResponses(
+        @PathVariable(STUDY_ID) studyId: UUID,
+        @PathVariable(PARTICIPANT_ID) participantId: String,
+        @PathVariable(QUESTIONNAIRE_ID) questionnaireId: UUID,
+        @RequestBody responses: List<QuestionnaireResponse>
+    ): OK {
+        surveysService.submitQuestionnaireResponses(studyId, participantId, questionnaireId, responses)
+        return OK()
+    }
+
+    override fun getQuestionnaireResponses(
+        studyId: UUID,
+        questionnaireId: UUID,
+        fileType: FileType
+    ): Iterable<Map<String, Any>> {
+        ensureReadAccess(AclKey(studyId))
+        return downloadService.getQuestionnaireResponses(studyId, questionnaireId)
+    }
+
+    @GetMapping(
+        path = [STUDY_ID_PATH + QUESTIONNAIRE_PATH + QUESTIONNAIRE_ID_PATH + DATA_PATH],
+        produces = [MediaType.APPLICATION_JSON_VALUE]
+    )
+    fun downloadQuestionnaireResponses(
+        @PathVariable(STUDY_ID) studyId: UUID,
+        @PathVariable(QUESTIONNAIRE_ID) questionnaireId: UUID,
+        @RequestParam(value = TYPE) fileType: FileType,
+        httpServletResponse: HttpServletResponse
+    ): Iterable<Map<String, Any>> {
+
+        val data = getQuestionnaireResponses(studyId, questionnaireId, fileType)
+        //TODO: rename file
+        val fileName = "Questionnaire_${questionnaireId}_${LocalDate.now().format(DateTimeFormatter.BASIC_ISO_DATE)}"
```

Comment:
I think it would be better if the file name was passed along as a query parameter instead of being hardcoded in the controller as it gives the users control over the file name.

Reply:
I think it's better to make the file name an optional parameter, and default to the hardcoded name if caller doesn't specify.